test/controllers/users_controller_test.js

```diff
@@ -241,6 +241,101 @@ describe('Endpoints exists for users', () => {
     });
   });
 
+  describe('Change user info', () => {
+    let userToken = null;
+    before((done) => {
+      chai.request(usersRoutes)
+        .post('/users')
+        .set('content-type', 'application/json')
+        .send({
+          email: 'userToBeChanged@email.com',
+          name: 'userToBeChanged',
+          password: 'passwordToBeChanged',
+        })
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(200);
+          chai.request(usersRoutes)
+            .post('/users/login')
+            .set('content-type', 'application/json')
+            .send({
+              email: 'userToBeChanged@email.com',
+              password: 'passwordToBeChanged',
+            })
+            .end((loginErr, loginRes) => {
+              should.not.exist(loginErr);
+              loginRes.should.have.status(200);
+              should.exist(loginRes.body.token);
+              userToken = loginRes.body.token;
+              done();
+            });
+        });
+    });
+
+    it('changes the user name successfully', (done) => {
+      chai.request(usersRoutes)
+        .put('/users/user_info')
+        .set('content-type', 'application/json')
+        .set('token', userToken)
+        .send({ name: 'New User Name' })
+        .end((changeNameErr, changeNameRes) => {
+          should.not.exist(changeNameErr);
+          changeNameRes.should.have.status(200);
+          chai.request(usersRoutes)
+            .get('/users/user_info')
+            .set('content-type', 'application/json')
+            .set('token', userToken)
+            .end((userInfoErr, userInfoRes) => {
+              should.not.exist(userInfoErr);
+              userInfoRes.should.have.status(200);
+              expect(userInfoRes.body.name).to.equal('New User Name');
+              done();
+            });
+        });
+    });
+    it('changes the user password successfully', (done) => {
+      chai.request(usersRoutes)
+        .put('/users/user_info')
+        .set('content-type', 'application/json')
+        .set('token', userToken)
+        .send({ password: 'New Password' })
+        .end((changePasswordErr, changePasswordRes) => {
+          should.not.exist(changePasswordErr);
+          changePasswordRes.should.have.status(200);
+          async.parallel({
+            loginWithNewPassword: async.reflect((callback) => {
```

Comment:
What about separating the test cases where the old password was unsuccessful and where the new one was successful?

Reply:
I don't think that's necessary - because both conditions need to be true at the same time for the password change to be considered successful.

Reply:
That's fair what about just an error message which shows which part of the test failed then